ultravox/tools/ds_tool/ds_tool.py

```diff
@@ -223,6 +226,127 @@ def __post_init__(self):
             self.upload_split = self.dataset_split
 
 
+class DatasetChunkProcessor:
+    args: DatasetToolArgs
+    cache_dir: str = ".cache/ds_tool/processed_datasets"
+    chunks_not_uploaded: List[Tuple[int, int]] = []
+    total_samples_processed: Dict[str, int] = defaultdict(int)
+
+    def __init__(self, args: DatasetToolArgs):
+        self.args = args
+
+    def process_and_upload_split_rescursive(
+        self,
+        split_name: str,
+        ds_split: datasets.Dataset,
+        start_index: int,
+        end_index: int,
+    ):
+        original_chunk_size = end_index - start_index
```

Comment:
`original_` in `original_chunk_size` implies you have a new `chunk_size`, but you are not really changing `chunk_size` here. You are simply splitting a dataset into chunks and do it once.

Reply:
Think we resolved this over call as well.